integration_test/ops_agent_test.go

```diff
@@ -3069,6 +3085,124 @@ metrics:
+	gce.RunForEachPlatform(t, func(t *testing.T, platform string) {
+		t.Parallel()
+		onlyReadScopes := strings.Join([]string{
+			"https://www.googleapis.com/auth/monitoring.read",
+			"https://www.googleapis.com/auth/logging.read",
+			"https://www.googleapis.com/auth/devstorage.read_write",
+		}, ",")
+		ctx, logger, vm := agents.CommonSetupWithExtraCreateArguments(t, platform, []string{"--scopes", onlyReadScopes})
+
+		if !gce.IsWindows(vm.Platform) {
+			packages := []string{"netcat"}
+			err := agents.InstallPackages(ctx, logger.ToMainLog(), vm, packages)
+			if err != nil {
+				t.Fatalf("failed to install %v with err: %s", packages, err)
+			}
+		}
+
+		if _, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", listenToPortForPlatform(vm.Platform)); err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(time.Minute)
+
+		if err := setupOpsAgent(ctx, logger, vm, ""); err != nil {
+			t.Fatal(err)
+		}
+
+		cmdOut, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", getRecentServiceOutputForPlatform(vm.Platform))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !strings.Contains(cmdOut.Stdout, "Network Check - Result: PASS") {
+			t.Errorf("expected network check to pass")
+		}
+		if !strings.Contains(cmdOut.Stdout, "API Check - Result: FAIL") {
+			t.Errorf("expected api check to fail")
+		}
+		if !strings.Contains(cmdOut.Stdout, "Ports Check - Result: FAIL") {
+			t.Errorf("expected ports check to fail")
+		}
+	})
+}
+
+func TestNetworkHealthCheck(t *testing.T) {
+	t.Parallel()
+	gce.RunForEachPlatform(t, func(t *testing.T, platform string) {
+		t.Parallel()
+		ctx, logger, vm := agents.CommonSetup(t, platform)
+
+		if err := setupOpsAgent(ctx, logger, vm, ""); err != nil {
+			t.Fatal(err)
+		}
+
+		cmdOut, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", getRecentServiceOutputForPlatform(vm.Platform))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if !strings.Contains(cmdOut.Stdout, "Network Check - Result: PASS") ||
+			!strings.Contains(cmdOut.Stdout, "API Check - Result: PASS") ||
+			!strings.Contains(cmdOut.Stdout, "Ports Check - Result: PASS") {
+			t.Errorf("expected all health checks to pass")
+		}
+
+		if _, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", stopCommandForPlatform(vm.Platform)); err != nil {
+			t.Fatal(err)
+		}
+
+		// Setting deny egress firewall rule. Waiting to changes to propagate and service output is farther in past.
+		if _, err := gce.AddTagToVm(ctx, logger.ToMainLog(), vm, []string{gce.DenyEgressTrafficTag}); err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(3 * time.Minute)
+
+		if _, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", startCommandForPlatform(vm.Platform)); err != nil {
+			t.Fatal(err)
+		}
+
+		cmdOut, err = gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", getRecentServiceOutputForPlatform(vm.Platform))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !strings.Contains(cmdOut.Stdout, "Network Check - Result: FAIL") {
+			t.Errorf("expected network check to fail")
```

Comment:
It's a good practice to put the expected values vs what you got. Makes it easier to debug:
t.Errorf("expected network check to fail. Got  value %v; expected to have %q", ...)

Reply:
Aside from the wording, I think this should say `expected network check to pass` instead of `expected network check to fail`?

Reply:
Could also refactor this a bit as:

```go
checkFunc := func(class string, expected string) {
	if !strings.Contains(cmdOut.Stdout, healthCheckResultMessage(class, expected)) {
		t.Errorf("expected %s check to %s", class, expected")
	}
}

checkFunc("Network", "PASS")
checkFunc("API", "ERROR")
checkFunc("Ports", "PASS")
```

Reply:
Thank you! This is a very good idea. I did an initial update for this on https://github.com/GoogleCloudPlatform/ops-agent/pull/1051/commits/d64ecc3c113b4da2afd047f60cf53402fbad9277 . I'm defining `checkFunc` multiple times though, not sure how good is this.